client/starwhale/utils/fs.py

```diff
@@ -164,14 +164,17 @@ def move_dir(src: Path, dest: Path, force: bool = False) -> t.Tuple[bool, str]:
         return True, f"{src} move to {dest}"
 
 
-def extract_tar(tar_path: Path, dest_dir: Path, force: bool = False) -> None:
+def extract_tar(
+    tar_path: Path, dest_dir: Path, force: bool = False, clear_dest: bool = True
+) -> None:
     if not tar_path.exists():
         raise NotFoundError(tar_path)
 
     if dest_dir.exists() and not force:
         raise ExistedError(str(dest_dir))
 
-    empty_dir(dest_dir)
+    if clear_dest:
```

Comment:
what's the difference between `force=True` and `clear_dest=True`?

if the caller config `force=True`, it can clear dest dir empty?

Reply:
The option of ’force‘ is to say whether to forcibly decompress tar to the target dir, but ’clear_dest‘ is to say whether to clear the target dir first when decompressing. There is a problem that when the dir where the tar is located is same with the target dir, it will cause the tar to be cleared

Reply:
done, only use force